File: packages/formbricks/ios/Plugin/Formbricks.swift
import Foundation
import FormbricksSDK

@objc public class Formbricks: NSObject {
    @objc public func logout() {
        FormbricksSDK.Formbricks.logout()
    }

    @objc public func setAttribute(_ options: SetAttributeOptions) {
        FormbricksSDK.Formbricks.setAttribute(options.getValue(), forKey: options.getKey())
    }

    @objc public func setAttributes(_ options: SetAttributesOptions) {
        let attributes = options.getAttributes().mapValues { AttributeValue.string($0) }
        FormbricksSDK.Formbricks.setAttributes(attributes)
    }

    @objc public func setLanguage(_ options: SetLanguageOptions) {
        FormbricksSDK.Formbricks.setLanguage(options.getLanguage())
    }

    @objc public func setUserId(_ options: SetUserIdOptions) {
        FormbricksSDK.Formbricks.setUserId(options.getUserId())
    }

    @objc public func setup(_ options: SetupOptions) {
        let config = FormbricksConfig.Builder(appUrl: options.getAppUrl(), environmentId: options.getEnvironmentId())
            .build()
        FormbricksSDK.Formbricks.setup(with: config)
    }

    @objc public func track(_ options: TrackOptions) {
        FormbricksSDK.Formbricks.track(options.getAction())
    }
}

File: packages/formbricks/android/src/main/java/io/capawesome/capacitorjs/plugins/formbricks/Formbricks.kt
package io.capawesome.capacitorjs.plugins.formbricks

import androidx.fragment.app.FragmentManager
import com.formbricks.android.helper.FormbricksConfig
import com.formbricks.android.model.user.AttributeValue
import io.capawesome.capacitorjs.plugins.formbricks.classes.options.SetAttributeOptions
import io.capawesome.capacitorjs.plugins.formbricks.classes.options.SetAttributesOptions
import io.capawesome.capacitorjs.plugins.formbricks.classes.options.SetLanguageOptions
import io.capawesome.capacitorjs.plugins.formbricks.classes.options.SetUserIdOptions
import io.capawesome.capacitorjs.plugins.formbricks.classes.options.SetupOptions
import io.capawesome.capacitorjs.plugins.formbricks.classes.options.TrackOptions
import com.formbricks.android.Formbricks as FormbricksSdk

class Formbricks(private val plugin: FormbricksPlugin) {

    fun logout() {
        FormbricksSdk.logout()
    }

    fun setAttribute(options: SetAttributeOptions) {
        FormbricksSdk.setAttribute(options.value, options.key)
    }

    fun setAttributes(options: SetAttributesOptions) {
        val attributes = options.attributes.mapValues { AttributeValue.string(it.value) }
        FormbricksSdk.setAttributes(attributes)
    }

    fun setLanguage(options: SetLanguageOptions) {
        FormbricksSdk.setLanguage(options.language)
    }

    fun setUserId(options: SetUserIdOptions) {
        FormbricksSdk.setUserId(options.userId)
    }

    fun setup(options: SetupOptions) {
        val fragmentManager: FragmentManager = plugin.activity.supportFragmentManager
        val config = FormbricksConfig.Builder(options.appUrl, options.environmentId)
            .setFragmentManager(fragmentManager)
            .build()
        FormbricksSdk.setup(plugin.context, config)
    }

    fun track(options: TrackOptions) {
        FormbricksSdk.track(options.action)
    }
}
